delay.go

```diff
@@ -6,24 +6,22 @@ import (
 	"time"
 )
 
-var sharedRNG = rand.New(rand.NewSource(time.Now().UnixNano()))
+var sharedRealSleeper = NewRealSleeper()
```

Comment:
This is a stylistic thing but I think I'd just name these like sleeper and NewSleeper(). I would put the onus on the tests to name things in a 'test' obvious way.

Reply:
Also should we bother with a shared sleeper? The cached RNG is probably expensive, but this sleeper is for sure not. I don't know if it's confusing or not.